src/valkey-benchmark.c

```diff
@@ -393,18 +394,23 @@ static void resetClient(client c) {
     c->pending = config.pipeline;
 }
 
-static void randomizeClientKey(client c) {
-    size_t i;
-
-    for (i = 0; i < c->randlen; i++) {
+static void generateClientKey(client c) {
+    static _Atomic size_t seq_key = 0;
+    for (size_t i = 0; i < c->randlen; i++) {
         char *p = c->randptr[i] + 11;
-        size_t r = 0;
-        if (config.randomkeys_keyspacelen != 0) r = random() % config.randomkeys_keyspacelen;
-        size_t j;
+        size_t key = 0;
+        if (config.keyspacelen != 0) {
+            if (config.sequential_replacement) {
+                key = atomic_fetch_add_explicit(&seq_key, 1, memory_order_relaxed);
```

Comment:
To guarantee that multiple threads get unique values here, can we really use relaxed menory ordering? Can you explain why this is enough?

I would guess we need sequential ordering.

Reply:
This is difficult stuff. I tried reading these two pages:

* https://en.cppreference.com/w/c/atomic/atomic_fetch_add
* https://en.cppreference.com/w/c/atomic/memory_order

Now I believe `memory_order_relaxed` is enough. The numbers may not be returned in right order according to the clock but I believe every number will be returned exactly once, because the function is atomic.